### DISCUSSION TITLE
Can rtl_433 convert to .wav?

### DISCUSSION THREAD

**Discussion by Crsarmv7l** [merbanan/rtl_433#2410] [Q&A] [answered]
Hopefully a simple question, can rtl_433 handle .wav files and convert to them? In the wiki it says they are "rare" but does not say if it can do so.

**Comment by zuckschwerdt** [comment, marked as answer]
No. Supporting [Broadcast Wave Format](https://en.wikipedia.org/wiki/Broadcast_Wave_Format) could be an option.
But it would be too easy to confuse FM or AM data ("audio" wav) with baseband data (I/Q wav) and a bad user experience.
Also the question is why? Where do you get wav files from? To what end are you hoping to process wav with rtl_433?

**Reply by Crsarmv7l** [reply]
For a little bit, I have been using rtl_433 src to recreate the bitstream of signals from scratch with python, then using Code_gen with the appropriate timings to create the .cu8 which works really well.

Since I was already working in python, I picked up a Yardstick One transceiver a while back. I learned it isn't a great receiver, so I am looking to utilize an rtlsdr in conjunction with the YS1.

There is one set of tools for the YS1 called ooktools which can pull the binary data and the baud from (ook) wave files, presumably recorded in gqrx.

So I was looking to take a .cu8 convert to .ook then to wave, tx with the YS1 and see if it shows up correctly when received by an rtlsdr.

Perhaps a better way would be for me to inspect a .ook file as that likely contains everything I need anyway, and I apologize for the disjointed idea. It was a late night thought and I haven't really dug into the YS1 very deeply as of yet.

**Reply by zuckschwerdt** [reply]
That's quite a stack there. When you say "(ook) wave files" that likely refers to AM demodulated data, i.e. a file at audio rates (44 kHz) and not baseband data. rtl_433 can parse and create that data ("am" dumper / "am" reader, see `rtl_433 -r` and `rtl_433 -w` for help), but it will be raw files and you would need to add a wav header with e.g. `sox`.

The CCx work in OOK data, so if you already have "pulse text" (.ook) it should be much simpler to work with.

**Reply by Crsarmv7l** [reply]
Yep, I am familiar with converting to .ook, I will take a look at what would be required with sox. I have used it before but I am not very familiar with it. Thanks for the tip

**Comment by merbanan** [comment]
Can you attach some of the work files you use? It should be easy to convert .cu8 to wav or bwav.

**Reply by Crsarmv7l** [reply]
I didn't have any specific file in mind, as more a proof of concept I was going to use one of the rtl test files. I wanted to see if it was possible before getting into doing it and developing a workflow

**Reply by Crsarmv7l** [reply]
As an example though, here is pulse data (.ook) from my Nexus-TH weather station (had to zip due to github restrictions)
[nexus.zip](https://github.com/merbanan/rtl_433/files/10913294/nexus.zip)

**Comment by Crsarmv7l** [comment]
Closing, I am an idiot. No sense jumping through all these hoops when Universal Radio Hacker allows the Yardstick 1 to transmit a signal from the generate tab with the plugin enabled. Simply import a signal into URH, crop (or dont), populate the generate table and modify (or dont) and click "Send via Yardstick 1."

So capture via rtl_433, and can send via YS1. I asked them to add it to the URH_cli, which would be really nice.

**Comment by Crsarmv7l** [comment]
Also as a side note, URH can save as .wav anyway. Being able to do it via cli would be ideal, but maybe in the future

**Reply by zuckschwerdt** [reply]
we might add wav / bmf support some day, but we really need to be careful to guess some metadata (maybe based on the sample rate): am/fm/iq all those could be wav, see also: `rtl_433 -r`
